wagtail/signal_handlers.py

```diff
@@ -83,6 +83,9 @@ def update_reference_index_on_save(instance, **kwargs):
             break
 
         instance = getattr(instance, parental_keys[0].name)
+        if instance is None:
```

Comment:
Should we rename `instance` to `parent`? This name might be confusing.

Or am I missing something?

Reply:
This code is inside a loop where the parent from the current iteration becomes the `instance` for the next.

We could do something like:
```
        parent = getattr(instance, parental_keys[0].name)
        if parent is None:
            # there is no valid object to record references against
            return
        instance = parent
```

but I think the comment already provides the necessary clarification.